scripts/prebuild.js

```diff
@@ -0,0 +1,98 @@
+const spawn = require('child_process').spawn;
+const globby = require('globby');
+const chalk = require('chalk');
+
+/** Execute the `cmd` in a new process. */
+const exec = (cmd) => {
+    return new Promise((resolve, reject) => {
+        console.log(chalk.green(`  ${cmd}`));
+        const command = spawn(cmd, [], {
+            shell: true,
+            stdio: 'inherit'
+        });
+
+        command.on('error', (err) => {
+            return reject(err);
+        });
+
+        command.on('exit', (code) => {
+            if (code !== 0) {
+                return reject(new Error('NoExitCodeZero'));
+            }
+
+            return resolve(true);
+        });
+
+    });
+};
+
+/**
+ * Searches all the `package.json` under `packages` and returns
+ * an array with the path and the content of each one.
+ * @returns { {packagePath: string, content: string}[] } The packages information
+ */
+const getPackages = async () => {
+    const packagesPaths = await globby([
+        `packages/*/package.json`,
+        `!packages/**/dist/**/*`,
+        `!packages/**/node_modules/**/*` // needed when we are inside a package like extension-vscode
+    ], {
+        absolute: true,
+        gitignore: true
+    });
+
+    const packages = packagesPaths.map((packagePath) => {
+        try {
+            const content = require(packagePath);
+
+            return {
+                content,
+                packagePath
+            };
+        } catch (e) {
+            // This happens if we are creating a new package and switching branches. We can ignore
```

Comment:
Is there a specific error message we could check for here? E.g. `Cannot find module '${packagePath}'`? I could see this potentially getting triggered due to a file permissions issue as well in which case we wouldn't want to ignore.

Reply:
In my case I wanted to ignore the error and continue the execution, that's why I didn't say anything. If we don't want to hide it, then let's remove the `try/catch`. The exception was clear enough to know what was going on.